File: database-mock.ts
export function connectFromRow(row: [number, string, { loose: boolean, lazy: boolean }]) {
    const options = row[2];
    const loose = options.loose = options.lazy || false;
    const timeout = row[0] * (loose ? 10 : 1);
    connectLegacy(row[1], { timeout, loose });
}
export function connectLegacy(address: string, options: { timeout?: number, loose: boolean } = { loose: false }): void {
}
export function connect(address: string, timeout: number, loose: boolean): void {
    // TODO: Actually connect, starting with making this call work: 
    // intended usage:  connect('https://some.site', timeout, /*loose*/ true)
}
/** Somehow copilot knew about the uses of findRow elsewhere in my project when suggesting this signature. */
export function findRow(table: string, conditions: [string, string][], order?: string, limit?: number): string[] {
    // TODO: Actually find row
    // (I don't know how much real code I need here to make this example work)
    return []
}

File: database_mock.py
from typing import Optional
def connect(address: str, timeout: int, loose: bool) -> None:
    # TODO: actually connect
    ...
def findRow(table: str, conditions: list[tuple[str, str]], order: Optional[str], limit: Optional[int]) -> list[str]:
    # TODO: Actually find row
    # (I don't know how much real code I need here to make this example work)
    connect('192.168.0.1', 1000, loose=True)
    ...
def connectFromRow(row: tuple[int, str, dict[str, bool]]):
    options = row[2];
    loose = options['loose'] = options['lazy'] or False
    timeout = row[0] * (10 if loose else 1);
    connect(row[1], timeout, loose);
def connectLegacy(address: str, timeout: Optional[int], loose: bool = False) -> None:
    # In Python, this is just an alias for connect with some defaults
    connect(address, timeout or 1000, loose)